[CI] ReindexDataStreamStatusTests testEqualsAndHashcode failing

**Build Scans:**
- [elasticsearch-periodic #5439 / concurrent-search-tests](https://gradle-enterprise.elastic.co/s/4jibgr2g2zrf6)
- [elasticsearch-periodic-platform-support #5428 / oraclelinux-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/ixbwcgy5fdu7g)
- [elasticsearch-pull-request #47333 / part-2](https://gradle-enterprise.elastic.co/s/uzm6phbq74udm)
- [elasticsearch-pull-request #47320 / part-2](https://gradle-enterprise.elastic.co/s/q7dso3qeeer5m)
- [elasticsearch-pull-request #47311 / part-2](https://gradle-enterprise.elastic.co/s/x6zbil3mkhmbg)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:migrate:test" --tests "org.elasticsearch.xpack.migrate.task.ReindexDataStreamStatusTests.testEqualsAndHashcode" -Dtests.seed=F2E54638D68B0B5D -Dtests.jvm.argline="-Des.concurrent_search=true" -Dtests.locale=oc-ES -Dtests.timezone=Africa/Mbabane -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.migrate.task.ReindexDataStreamStatusTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testEqualsAndHashcode),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: max must be >= min: 101, 100
```

**Issue Reasons:**
- [main] 5 failures in test testEqualsAndHashcode (1.7% fail rate in 288 executions)
- [main] 3 failures in step part-2 (1.9% fail rate in 161 executions)
- [main] 3 failures in pipeline elasticsearch-pull-request (1.9% fail rate in 159 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -292,9 +292,6 @@ tests:
   issue: https://github.com/elastic/elasticsearch/issues/118914
 - class: org.elasticsearch.smoketest.SmokeTestMultiNodeClientYamlTestSuiteIT
   issue: https://github.com/elastic/elasticsearch/issues/118955
-- class: org.elasticsearch.xpack.migrate.task.ReindexDataStreamStatusTests
-  method: testEqualsAndHashcode
-  issue: https://github.com/elastic/elasticsearch/issues/118965
 - class: org.elasticsearch.repositories.blobstore.testkit.analyze.SecureHdfsRepositoryAnalysisRestIT
   issue: https://github.com/elastic/elasticsearch/issues/118970
 

